tests/DCPS/Compiler/anonymous_types/test.idl

```diff
@@ -0,0 +1,116 @@
+#include <tao/LongSeq.pidl>
+
+typedef short Short3Array[3];
+typedef short Short2x3Array[2][3];
+typedef double Double2x3x4Array[2][3][4];
+typedef string Str2x3x4x5Array[2][3][4][5];
+
+typedef Short3Array Short3Array_x2[2];
+
+struct StructOfMultiDimArray {
+  Str2x3x4x5Array str2x3x4x5arr;
+  Short3Array_x2 s3arr_x2arr;
+};
+
+typedef sequence<long> LongSequence;
+typedef sequence<long, 3> Long3Sequence;
+typedef string<100> BoundedString;
+
+typedef struct XYZ_ {
```

Comment:
Does this need to be "typedef struct" instead of just "struct" in order to test something in the compiler?
Otherwise just make it a normal struct.

Reply:
We have quite a few normal structs, StructOfMultiDimArray, EntityId_t, GUID_t, RelayHeader, and all the topic structs. Here I wanted to have some more coverage and to see where XYZ_ and XYZ appear in the generated code. 

I created struct NS and sequence<NS, 5> a_ns5seq to test a normal struct used in an anonymous field.